packages/client/src/clients/in-app-messages/message-client.ts

```diff
@@ -233,6 +238,30 @@ export class InAppMessagesClient {
     return paramsString ? `${baseKey}?${paramsString}` : baseKey;
   }
 
+  subscribe() {
+    this.unsub = this.channelClient.subscribe(this);
+  }
+
+  unsubscribe() {
+    this.channelClient.unsubscribe(this);
+  }
+
+  // This is a callback function that will be invoked when a new socket event
+  // relevant for this message client is received (and if subscribed).
+  async handleSocketEvent(payload: SocketEventPayload) {
+    switch (payload.event) {
+      case SocketEventType.MessageCreated:
+        return await this.fetch();
```

Comment:
Per my point in the Switchboard PR, no need to refetch here now!

Reply:
Replied in the switchboard PR thread - I did consider doing something smarter here with the in-app message included in the event payload instead of refetching but I realized it wasn't going to be as simple as I initially thought. 

For reference, more details in this thread: https://github.com/knocklabs/switchboard/pull/2963#discussion_r1803877703